sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequest.java

```diff
@@ -78,13 +79,17 @@ public static RntbdRequest decode(final ByteBuf in) {
     void encode(final ByteBuf out) {
 
         final int expectedLength = RntbdRequestFrame.LENGTH + this.headers.computeLength();
-        final int start = out.readerIndex();
+        checkState(out.readerIndex() == out.writerIndex());
```

Comment:
Suggestion: Can unrelated changes be  pushed as separate iteration?

Reply:
This is a change that I'll keep in. It is a potential bug that I caught while debugging. I must check the writerIndex, not the reader index. It is luck that we haven't had a problem here.

Reply:
I did remove the checkState line as this was a check to ensure I hadn't broken anything based on correcting the code. This verified that I wasn't depending on the bug.

Reply:
change is in my branch

Reply:
change committed and pushed.